docs/source/maint_processes.rst

```diff
@@ -0,0 +1,41 @@
+********************
```

Comment:
Does this belong in the docs? WE have CONTRIBUTING.md that has a lot of the current development flow documentation.

Reply:
Hm, you're right. But then, `CONTRIBUTING.md` grew too big. It started as a small document and evolved into a second documentation next to the actual readthedocs documentation (and the wiki). I don't think that's good for discoverability, so I'd opt for moving some of the CONTRIBUTING.md content also into the main docs. 

If others agree, we can do that all in one go here, or separately (or not at all?)

@cmarqu thoughts?

Reply:
We should at least leave a stub `CONTRIBUTING.md` pointing to the Sphinx doc section since that file is kinda standard and e.g. GitHub will check for it in https://github.com/cocotb/cocotb/community